ui/app/peers/create/[peerType]/helpers/my.ts

```diff
@@ -100,6 +105,117 @@ export const mysqlSetting: PeerSetting[] = [
     optional: true,
     tips: 'If not provided, host CA roots will be used.',
   },
+
+  {
+    label: 'Authentication type',
+    field: 'authType',
+    default: 'MYSQL_PASSWORD',
+    stateHandler: (value, setter) =>
+      setter((curr) => ({
+        ...(curr as MySqlConfig),
+        authType: mySqlAuthTypeFromJSON(value),
+      })),
+    type: 'select',
+    placeholder: 'Select authentication mechanism',
+    options: [
+      { value: 'MYSQL_PASSWORD', label: 'Password' },
+      { value: 'MYSQL_IAM_AUTH', label: 'AWS IAM Auth' },
+    ],
+    tips: 'AWS IAM Auth is supported for all Mirror Types for MySQL',
+  },
+  {
+    label: 'AWS IAM Auth Mechanism',
+    field: 'awsAuth.authType',
+    default: 'IAM_AUTH_AUTOMATIC',
+    stateHandler: (value, setter) =>
+      setter((curr) => {
+        let cfg = curr as MySqlConfig;
+        let awsAuth: AwsAuthenticationConfig = {
+          region: '',
+          authType: awsIAMAuthConfigTypeFromJSON(value),
+        };
+        switch (awsAuth.authType) {
+          case AwsIAMAuthConfigType.IAM_AUTH_STATIC_CREDENTIALS:
+            awsAuth = {
+              ...awsAuth,
+              staticCredentials: {
+                accessKeyId: '',
+                secretAccessKey: '',
+              },
+            };
+            break;
+          case AwsIAMAuthConfigType.IAM_AUTH_ASSUME_ROLE:
+            awsAuth = {
+              ...awsAuth,
+              role: {
+                assumeRoleArn: '',
+              },
+            };
+            break;
+        }
+        const newCfg = {
+          ...cfg,
+          awsAuth: awsAuth,
+        };
+        return newCfg;
+      }),
+    type: 'select',
+    placeholder: 'Select authentication mechanism',
+    options: [
+      { value: 'IAM_AUTH_AUTOMATIC', label: 'Automatic' },
+      { value: 'IAM_AUTH_STATIC_CREDENTIALS', label: 'Static Credentials' },
+      { value: 'IAM_AUTH_ASSUME_ROLE', label: 'IAM Role Chaining' },
+    ],
+    tips: 'Automatic uses the default AWS credentials provider chain. Static Credentials uses the static credentials you provide. IAM Role Chaining uses the role you provide to assume a different role (along with a chained role if provided).',
+  },
+  {
+    label: 'IAM Role to Assume',
+    field: 'awsAuth.role.assumeIamRole',
+    stateHandler: (value, setter) =>
+      setter((curr) => {
+        let mysqlConfig = curr as MySqlConfig;
+        mysqlConfig.awsAuth!.role!.assumeRoleArn = value as string;
+        return { ...mysqlConfig };
+      }),
+    tips: 'AWS IAM Role to assume.',
+  },
+  {
+    label: 'Chained IAM Role to Assume',
+    field: 'awsAuth.role.chainedIamRole',
+    stateHandler: (value, setter) =>
+      setter((curr) => {
+        let mysqlConfig = curr as MySqlConfig;
+        mysqlConfig.awsAuth!.role!.assumeRoleArn = value as string;
```

Comment:
I think this needs to be `chainedIamRole`

Reply:
Yes, I was just testing this and running into an issue